src/components/common/Layout/components/Account/components/Notifications/components/MessagingContext.tsx

```diff
@@ -0,0 +1,30 @@
+import { createContext, useContext } from "react"
+import { useWeb3InboxSubscription } from "./web3Inbox"
+import { useSubscribeXmtp, useXmtpAccessChecking } from "./xmtp"
+
+const messagingContext = createContext<
+  ReturnType<typeof useSubscribeXmtp> &
+    ReturnType<typeof useWeb3InboxSubscription> &
+    ReturnType<typeof useXmtpAccessChecking>
+>(undefined)
+
+export const useMessagingContext = () => useContext(messagingContext)
+
+export const MessagingWrapper = ({ children }) => {
```

Comment:
Please follow the naming conventions from our other contexts & rename this to `MessagingProvider`.
Or, I wonder if we actually need this context? I know we wanted to handle most of the logic here, but since we couldn't do everything what we wanted with XMTP, I think this is a bit overkill now. We can just call these 3 hooks where we need them instead ;)

Reply:
At first, it seemed like a good idea, but now I think it's overkill yes.